File: src/deposit.cpp
#include "include/deposit.h"

big_int deposit::bill_id = 0;

big_int deposit::get_cash_size() const {
  return cash_size_;
}

std::string deposit::get_bill_kind() const {
  return BILL_KIND_DEPOSIT;
}

big_int deposit::get_income_potential_size() const {
  return income_potential_size_;
}

big_int deposit::get_bill_id() const {
  return bill_id_;
}

big_int deposit::get_person_id() const {
  return person_id_;
}

void deposit::set_cash_size(const big_int& cash_size) {
  cash_size_ = cash_size;
}

void deposit::set_income_potential_size(const big_int& income_potential_size) {
  income_potential_size_ = income_potential_size;
}

void deposit::set_person_id(const big_int& person_id) {
  person_id_ = person_id;
}

void deposit::update() {
  income_potential_size_ = cash_size_ / 100 * 10;
}

void deposit::assign_id() {
  bill_id_ = bill_id++;
}

File: MIPT_bank_app1/MIPT_bank_app/app/src/main/java/com/example/mipt_bank_app/Deposit.java
package com.example.mipt_bank_app;


public class Deposit extends IBill {
    @Override
    public String getBillKind() {
        return bill_kind_;
    }

    @Override
    public int getCashSize() {
        return cash_size_;
    }

    public int getIncomePotentialSize() {
        return income_potential_size_;
    }

    @Override
    public int getBillId() {
        return bill_id_;
    }

    @Override
    public int getPersonId() {
        return person_id_;
    }

    @Override
    public void setCashSize(int cash_size) {
        cash_size_ = cash_size;
    }

    @Override
    public void setBillId(int id) {
        bill_id_ = id;
    }

    public void setIncomePotentialSize(int income_potential_size) {
        income_potential_size_ = income_potential_size;
    }

    @Override
    public void setPersonId(int person_id) {
        person_id_ = person_id;
    }

    void update() {
        income_potential_size_ = cash_size_ / 100 * 10;
    }

    @Override
    public void assignId() {

    }

    private String bill_kind_ = SupConstants.BILL_KIND_DEPOSIT;
    private int income_potential_size_;

}
